modernizer-maven-plugin/src/main/java/org/gaul/modernizer_maven_plugin/output/CodeClimateOutputer.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2014-2022 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.modernizer_maven_plugin.output;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.reflect.TypeToken;
+
+import org.gaul.modernizer_maven_plugin.Violation;
+
+public final class CodeClimateOutputer implements Outputer {
+    public static final String DEFAULT_FILENAME = "code-quality.json";
+
+    private final String outputFile;
+    private final Severity severity;
+
+    public CodeClimateOutputer(String outputFile, Severity severity) {
+        this.outputFile = outputFile;
+        this.severity = severity;
+    }
+
+    @Override
+    public void output(List<OutputEntry> entries) throws IOException {
+        List<Entry> toOutput = new ArrayList<>(entries.size());
+        for (OutputEntry entry : entries) {
+            Location location = new Location(
+                    entry.getFileName(),
+                    new Location.Lines(entry.getOccurrence().getLineNumber()));
+            Violation violation = entry.getOccurrence().getViolation();
+            toOutput.add(new Entry(
+                    violation.getComment(),
+                    violation.getName(),
+                    Integer.toString(entry.hashCode()),
+                    severity,
+                    location));
+        }
+        TypeToken<List<Entry>> type = new TypeToken<List<Entry>>() {
+        };
+        OutputStreamWriter writer = new OutputStreamWriter(
```

Comment:
Should this be in try-with-resources since you bumped the Java version?

Reply:
That it should.